### ISSUE TITLE
Valet share: Not working with SSL

### ISSUE THREAD

**Issue by hitech95** [cpriego/valet-linux#107]
Looks like that after sharing a site the tunnel redirect to a wrong port on the local server. If is SSL ngrok reirect the traffic to example.dev:80 instead of 443

```
Forwarding                    http://3bb56ffd.ngrok.io -> example.dev:80                                                                                                                       
Forwarding                    https://3bb56ffd.ngrok.io -> example.dev:80  
```

**Comment by TheNodi** [comment]
This is a known issue, unfortunately I haven't had the time to fix it yet. If anyone wants to help, feel free to do it.

**Comment by hitech95** [comment]
No Idea on where to start, if you give me a start point I will help you :+1:

**Comment by TheNodi** [comment]
Ngrok port is taken from [config.json](https://github.com/cpriego/valet-linux/blob/master/valet#L83) to allow the users to customize their configurations. This is implemented only half-way, the http port is configurable, https is not. 

I wanted to implement a custom setting for https port, and when sharing retrieve the port based on if the site is secure or not.

The line I linked before should become something like:
```bash
PORT=$(php "$DIR/cli/valet.php" current-port "$HOST")
```
This way is up to php determine if the site is secured and returning the correct port.
It should also remove the need for `jq` (maybe, need to check). 

I need to test if ngrok can tunnel a self-signed https site.

If you have any suggestion I would love to hear them, this weekend I'll try to write something out.

**Comment by TheNodi** [comment]
I tested out ngrok with https site and I concluded that it's not possible unless you buy a custom subdomain, or do some tricky hacks using a raw tcp tunnel and generating a custom `something.ngrok.io` ssl certificate. This is far more complex than what I expected and beyond valet. 

I'm gonna PR an error message when trying to share a secured site.

### TL;DR
You can't share a secure valet domain. Unsecure it and use the https link provided by ngrok to access it.

**Comment by yamenarahman** [comment]
https://github.com/laravel/valet/issues/631 Would this help?
